Nterl0k - T1033 Query.exe usage on remote devices.

nterl0k opened the pull request:
### Details

More specific detection of Query.exe being used against remote devices.

Pending https://github.com/splunk/attack_data/pull/931

### Checklist

- [ ] Validate name matches `<platform>_<mitre att&ck technique>_<short description>` nomenclature
- [ ] [CI/CD](https://github.com/splunk/security_content/actions) jobs passed ✔️ 
- [ ] Validated SPL logic.
- [ ] Validated tags, description, and how to implement.
- [ ] Verified references match analytic.
- [ ] Confirm updates to lookups are handled properly.

### Notes For Submitters and Reviewers

- If you're submitting a PR from a fork, ensuring the box to allow updates from maintainers is checked will help speed up the process of getting it merged.
- Checking the output of the `build` CI job when it fails will likely show an error about what is failing. You may have a very descriptive error of the specific field(s) in the specific file(s) that is causing an issue. In some cases, its also possible there is an issue with the YAML. Many of these can be caught with the pre-commit hooks if you set them up. These errors will be less descriptive as to what exactly is wrong, but will give you a column and row position in a specific file where the YAML processing breaks. If you're having trouble with this, feel free to add a comment to your PR tagging one of the maintainers and we'll be happy to help troubleshoot it.
- Updates to existing lookup files can be tricky, because of how Splunk handles application updates and the differences between existing lookup files being updated vs new lookups. You can read more [here](https://docs.splunk.com/Documentation/SplunkCloud/8.2.2203/Admin/PrivateApps#Manage_lookups_in_Splunk_Cloud_Platform) but the short version is that any changes to lookup files need to bump the datestamp in the lookup CSV filename, and the reference to it in the YAML needs to be updated.

nterl0k committed:
Add files via upload

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+date: '2025-01-06'
+author: Steven Dick
+status: production
+type: TTP
```
Having query simply execute is not TTP worthy imo.

```suggestion
type: Anomaly

```

nterl0k replied:
Could vary depending on environment, but I have no issues setting it to RBA only.

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+description: The following analytic detects the execution of `query.exe` with command-line arguments aimed at discovering data on remote devices. It leverages data from Endpoint Detection and Response (EDR) agents, focusing on process names and command-line executions. This activity is significant as adversaries may use `query.exe` to gain situational awareness and perform Active Directory discov ...
+data_source: 
+- Sysmon Event ID 1
+- Windows Security Event ID 4688
```
```suggestion
- Windows Security Event ID 4688
- CrowdStrike ProcessRollup2
```

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+  - Active Directory Discovery
+  asset_type: Endpoint
+  confidence: 75
+  impact: 60
```
A single event for query is medium severity at best

```suggestion
  impact: 49

```

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+  analytic_story: 
+  - Active Directory Discovery
+  asset_type: Endpoint
+  confidence: 75
```
it should be medium https://github.com/splunk/security_content/wiki/6.1-%E2%80%90-How-are-risk-score-calculated-for-RBA

```suggestion
  confidence: 49

```

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+  - Processes.process
+  - Processes.process_name 
+  - Processes.parent_process_name
+  risk_score: 45
```
```suggestion
  risk_score: 24

```

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+  asset_type: Endpoint
+  confidence: 75
+  impact: 60
+  message: The user $user$ ran the Query command on a remote system from $dest$
```
```suggestion
  message: The user $user$ ran the Query command to query information from a remote system from $dest$

```

nasbench committed:
Merge branch 'develop' into nterl0k-t1033-query-remote-usage

nasbench commented on detections/endpoint/windows_system_remote_discovery_with_query.yml:
```diff
@@ -0,0 +1,57 @@
+name: Windows System Remote Discovery With Query
```
Just noticed that we also have ad03bfcf-8a91-4bc2-a500-112993deba87 focusing on user discovery. In an edge case where a user would use 

`query user /sever:` they would both match creating an inflated score.

We perhaps need to exclude the "/server" flag from the other rule to focus only on local queries

nterl0k replied:
Agreed, I noticed the duplicate firing in our environment and choose to alter risk scoring on the user discovery correlation accordingly. 

I can agree this is a bit of an overlap in coverage with the existing detection, but feel the lateral movement aspect adds more risk.

nasbench replied:
Altered the detection of both rules to not overlap.

nterl0k committed:
Update detections/endpoint/windows_system_remote_discovery_with_query.yml

Good suggestion

Co-authored-by: Nasreddine Bencherchali <monsteroffire2@gmail.com>

nterl0k committed:
Update detections/endpoint/windows_system_remote_discovery_with_query.yml

EID updates

Co-authored-by: Nasreddine Bencherchali <monsteroffire2@gmail.com>

nterl0k committed:
Apply suggestions from code review

Bulk commit on risk score and grammar

Co-authored-by: Nasreddine Bencherchali <monsteroffire2@gmail.com>

nterl0k committed:
Update windows_system_remote_discovery_with_query.yml

Updating with drilldowns

nterl0k committed:
Update windows_system_remote_discovery_with_query.yml

observables update

nterl0k commented:
This would pass if I didn't put the testing data in the wrong folder 

https://github.com/splunk/attack_data/pull/936

patel-bhavin committed:
Merge branch 'develop' into nterl0k-t1033-query-remote-usage

patel-bhavin committed:
Merge branch 'develop' into nterl0k-t1033-query-remote-usage

nterl0k committed:
Update windows_system_remote_discovery_with_query.yml

update to v5 yaml spec 
 update search yaml for better readability / remove single quote in SPL issues

nterl0k committed:
Merge branch 'splunk:develop' into nterl0k-t1033-query-remote-usage

patel-bhavin committed:
Merge branch 'develop' into nterl0k-t1033-query-remote-usage

nasbench committed:
update logic to avoid duplicate alerting

nasbench committed:
Merge branch 'develop' into nterl0k-t1033-query-remote-usage
